spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestSparkWriteConf.java

```diff
@@ -0,0 +1,197 @@
+    Assert.assertEquals(DistributionMode.HASH, writeConf.updateDistributionMode());
+    Assert.assertEquals(DistributionMode.HASH, writeConf.copyOnWriteMergeDistributionMode());
+    Assert.assertEquals(DistributionMode.HASH, writeConf.positionDeltaMergeDistributionMode());
+  }
+
+  @Test
+  public void testSparkWriteConfDistributionModeWithWriteOption() {
+    Table table = validationCatalog.loadTable(tableIdent);
+
+    Map<String, String> writeOptions =
+        ImmutableMap.of(SparkWriteOptions.DISTRIBUTION_MODE, DistributionMode.NONE.modeName());
+
+    SparkWriteConf writeConf = new SparkWriteConf(spark, table, writeOptions);
+    Assert.assertEquals(DistributionMode.NONE, writeConf.distributionMode());
+    Assert.assertEquals(DistributionMode.NONE, writeConf.deleteDistributionMode());
+    Assert.assertEquals(DistributionMode.NONE, writeConf.updateDistributionMode());
+    Assert.assertEquals(DistributionMode.NONE, writeConf.copyOnWriteMergeDistributionMode());
+    Assert.assertEquals(DistributionMode.NONE, writeConf.positionDeltaMergeDistributionMode());
+  }
+
+  @Test
+  public void testSparkWriteConfDistributionModeWithSessionConfig() {
+    withSQLConf(
+        ImmutableMap.of(SparkSQLProperties.DISTRIBUTION_MODE, DistributionMode.NONE.modeName()),
+        () -> {
+          Table table = validationCatalog.loadTable(tableIdent);
+
+          SparkWriteConf writeConf = new SparkWriteConf(spark, table, ImmutableMap.of());
+
+          Assert.assertEquals(DistributionMode.NONE, writeConf.distributionMode());
+          Assert.assertEquals(DistributionMode.NONE, writeConf.deleteDistributionMode());
+          Assert.assertEquals(DistributionMode.NONE, writeConf.updateDistributionMode());
+          Assert.assertEquals(DistributionMode.NONE, writeConf.copyOnWriteMergeDistributionMode());
+          Assert.assertEquals(
+              DistributionMode.NONE, writeConf.positionDeltaMergeDistributionMode());
+        });
+  }
+
+  @Test
+  public void testSparkWriteConfDistributionModeWithTableProperties() {
+    Table table = validationCatalog.loadTable(tableIdent);
+
+    table
+        .updateProperties()
+        .set(WRITE_DISTRIBUTION_MODE, WRITE_DISTRIBUTION_MODE_NONE)
+        .set(DELETE_DISTRIBUTION_MODE, WRITE_DISTRIBUTION_MODE_NONE)
+        .set(UPDATE_DISTRIBUTION_MODE, WRITE_DISTRIBUTION_MODE_NONE)
+        .set(MERGE_DISTRIBUTION_MODE, WRITE_DISTRIBUTION_MODE_NONE)
+        .commit();
+
+    SparkWriteConf writeConf = new SparkWriteConf(spark, table, ImmutableMap.of());
+    Assert.assertEquals(DistributionMode.NONE, writeConf.distributionMode());
+    Assert.assertEquals(DistributionMode.NONE, writeConf.deleteDistributionMode());
+    Assert.assertEquals(DistributionMode.NONE, writeConf.updateDistributionMode());
+    Assert.assertEquals(DistributionMode.NONE, writeConf.copyOnWriteMergeDistributionMode());
+    Assert.assertEquals(DistributionMode.NONE, writeConf.positionDeltaMergeDistributionMode());
```

Comment:
Nonblocking nit: I guess this is a case where we can have some kind of helper function to validate that all of the write conf options match the given distribution mode.